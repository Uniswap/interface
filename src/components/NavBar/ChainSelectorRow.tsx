import { useWeb3React } from '@web3-react/core'
import Loader from 'components/Loader'
import { getChainInfo } from 'constants/chainInfo'
import { SupportedChainId } from 'constants/chains'
import { CheckMarkIcon } from 'nft/components/icons'
import styled, { useTheme } from 'styled-components/macro'

const LOGO_SIZE = 20

const Container = styled.button`
  display: grid;
  background: none;
  grid-template-columns: min-content 1fr min-content;
  align-items: center;
  text-align: left;
  line-height: 24px;
  border: none;
  justify-content: space-between;
  padding: 10px 8px;
  cursor: pointer;
  border-radius: 12px;
  color: ${({ theme }) => theme.textPrimary};
  width: 240px;
  transition: ${({ theme }) => theme.transition.duration.medium} ${({ theme }) => theme.transition.timing.ease}
    background-color;

  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.sm}px`}) {
    width: 100%;
  }

  &:hover {
    background-color: ${({ theme }) => theme.backgroundOutline};
  }
`

const Label = styled.div`
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
`

const Status = styled.div`
  grid-column: 3;
  grid-row: 1;
  display: flex;
  align-items: center;
  width: ${LOGO_SIZE}px;
`

const ApproveText = styled.div`
  color: ${({ theme }) => theme.textSecondary};
  font-size: 12px;
  grid-column: 2;
  grid-row: 2;
`

const Logo = styled.img`
  height: ${LOGO_SIZE}px;
  width: ${LOGO_SIZE}px;
  margin-right: 12px;
`

export default function ChainSelectorRow({
  targetChain,
  onSelectChain,
  isPending,
}: {
  targetChain: SupportedChainId
  onSelectChain: (targetChain: number) => void
  isPending: boolean
}) {
  const { chainId } = useWeb3React()
  const active = chainId === targetChain
  const { label, logoUrl } = getChainInfo(targetChain)

  const theme = useTheme()

  return (
    <Container onClick={() => onSelectChain(targetChain)}>
      <Logo src={logoUrl} alt={label} />
      <Label>{label}</Label>
      {isPending && <ApproveText>Approve in wallet</ApproveText>}
      <Status>
        {active && <CheckMarkIcon width={LOGO_SIZE} height={LOGO_SIZE} color={theme.accentActive} />}
        {isPending && <Loader width={LOGO_SIZE} height={LOGO_SIZE} />}
      </Status>
    </Container>
  )
}
