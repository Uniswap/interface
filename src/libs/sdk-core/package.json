{
  "name": "@uniswap/sdk-core",
  "license": "MIT",
  "version": "3.0.0-alpha.1",
  "description": "⚒️ An SDK for building applications on top of Uniswap V3",
  "main": "dist/index.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "repository": "https://github.com/Uniswap/uniswap-sdk-core.git",
  "keywords": [
    "uniswap",
    "ethereum"
  ],
  "module": "dist/sdk-core.esm.js",
  "scripts": {
    "build": "tsdx build",
    "start": "tsdx watch",
    "test": "tsdx test",
    "prepublishOnly": "tsdx build"
  },
  "dependencies": {
    "@ethersproject/address": "^5.0.2",
    "big.js": "^5.2.2",
    "decimal.js-light": "^2.5.0",
    "jsbi": "^3.1.4",
    "tiny-invariant": "^1.1.0",
    "toformat": "^2.0.0"
  },
  "devDependencies": {
    "@types/big.js": "^4.0.5",
    "@types/jest": "^24.0.25",
    "tsdx": "^0.14.1"
  },
  "engines": {
    "node": ">=10"
  },
  "prettier": {
    "printWidth": 120,
    "semi": false,
    "singleQuote": true
  }
}
