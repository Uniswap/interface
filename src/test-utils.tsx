import { render, RenderOptions } from '@testing-library/react'
import React, { FC, ReactElement, ReactNode } from 'react'
import { Provider } from 'react-redux'
import store from 'state'
import ThemeProvider from 'theme'

const WithProviders: FC = ({ children }: { children?: ReactNode }) => {
  return (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: WithProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
