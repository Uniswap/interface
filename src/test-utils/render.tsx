import { MockedProvider } from '@apollo/client/testing'
import { i18n } from '@lingui/core'
import { I18nProvider } from '@lingui/react'
import { queries } from '@testing-library/dom'
import { render, renderHook, RenderHookOptions, RenderOptions } from '@testing-library/react'
import { DEFAULT_LOCALE } from 'constants/locales'
import { BlockNumberProvider } from 'lib/hooks/useBlockNumber'
import catalog from 'locales/en-US'
import { en } from 'make-plural/plurals'
import { ReactElement, ReactNode } from 'react'
import { QueryClient, QueryClientProvider } from 'react-query'
import { Provider } from 'react-redux'
import { HashRouter } from 'react-router-dom'
import store from 'state'
import ThemeProvider from 'theme'

i18n.load({
  [DEFAULT_LOCALE]: catalog.messages,
})
i18n.loadLocaleData({
  [DEFAULT_LOCALE]: { plurals: en },
})
i18n.activate(DEFAULT_LOCALE)
const MockedI18nProvider = ({ children }: any) => <I18nProvider i18n={i18n}>{children}</I18nProvider>

const queryClient = new QueryClient()

const WithProviders = ({ children }: { children?: ReactNode }) => {
  return (
    <MockedI18nProvider>
      <Provider store={store}>
        <QueryClientProvider client={queryClient}>
          <HashRouter>
            {/*
             * Web3Provider is mocked through setupTests.ts
             * To test behavior that depends on Web3Provider, use jest.unmock('@web3-react/core')
             */}
            <MockedProvider>
              <BlockNumberProvider>
                <ThemeProvider>{children}</ThemeProvider>
              </BlockNumberProvider>
            </MockedProvider>
          </HashRouter>
        </QueryClientProvider>
      </Provider>
    </MockedI18nProvider>
  )
}

type CustomRenderOptions = Omit<RenderOptions, 'wrapper'>
const customRender = (ui: ReactElement, options?: CustomRenderOptions) => {
  return render<typeof queries>(ui, { ...options, wrapper: WithProviders })
}

type CustomRenderHookOptions<Props> = Omit<RenderHookOptions<Props>, 'wrapper'>
const customRenderHook = <Result, Props>(
  hook: (initialProps: Props) => Result,
  options?: CustomRenderHookOptions<Props>
) => {
  return renderHook(hook, { ...options, wrapper: WithProviders })
}

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
