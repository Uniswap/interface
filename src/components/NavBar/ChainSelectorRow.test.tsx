import { ChainId, SUPPORTED_CHAINS } from '@uniswap/sdk-core'
import { render } from 'test-utils/render'

import ChainSelectorRow from './ChainSelectorRow'

describe('ChainSelectorRow', () => {
  SUPPORTED_CHAINS.forEach((chainId) => {
    it(`should match snapshot for chainId ${chainId}`, () => {
      const { container } = render(
        <ChainSelectorRow targetChain={chainId} onSelectChain={jest.fn()} isPending={false} disabled={false} />
      )
      expect(container).toMatchSnapshot()
    })
  })

  it('should be clickable when enabled', () => {
    const onSelectChain = jest.fn()
    const { getByTestId } = render(
      <ChainSelectorRow
        targetChain={ChainId.OPTIMISM}
        onSelectChain={onSelectChain}
        isPending={false}
        disabled={false}
      />
    )
    const button = getByTestId('Optimism-selector')
    button.click()
    expect(onSelectChain).toHaveBeenCalled()
  })

  it('should not be clickable when disabled', () => {
    const onSelectChain = jest.fn()
    const { getByTestId } = render(
      <ChainSelectorRow
        targetChain={ChainId.OPTIMISM}
        onSelectChain={onSelectChain}
        isPending={false}
        disabled={true}
      />
    )
    const button = getByTestId('Optimism-selector')
    button.click()
    expect(onSelectChain).not.toHaveBeenCalled()
  })
})
