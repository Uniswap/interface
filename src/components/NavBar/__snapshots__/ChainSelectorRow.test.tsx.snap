// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChainSelectorRow should match snapshot for chainId 1 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Ethereum-selector"
  >
    <img
      alt="Ethereum"
      class="c1"
      src="ethereum-logo.png"
    />
    <div
      class="c2"
    >
      Ethereum
    </div>
    <div
      class="c3"
    >
      <svg
        color="#FC72FF"
        fill="none"
        height="20"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.6693 5.33789L7.5026 14.3175L3.33594 10.2358"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </div>
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 5 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Görli-selector"
  >
    <img
      alt="Görli"
      class="c1"
      src="ethereum-logo.png"
    />
    <div
      class="c2"
    >
      Görli
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 10 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Optimism-selector"
  >
    <img
      alt="Optimism"
      class="c1"
      src="optimistic_ethereum.svg"
    />
    <div
      class="c2"
    >
      Optimism
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 56 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="BNB Chain-selector"
  >
    <img
      alt="BNB Chain"
      class="c1"
      src="bnb-logo.svg"
    />
    <div
      class="c2"
    >
      BNB Chain
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 137 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Polygon-selector"
  >
    <img
      alt="Polygon"
      class="c1"
      src="polygon-matic-logo.svg"
    />
    <div
      class="c2"
    >
      Polygon
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 420 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Optimism Görli-selector"
  >
    <img
      alt="Optimism Görli"
      class="c1"
      src="optimistic_ethereum.svg"
    />
    <div
      class="c2"
    >
      Optimism Görli
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 8453 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Base-selector"
  >
    <img
      alt="Base"
      class="c1"
      src="base_logo.svg"
    />
    <div
      class="c2"
    >
      Base
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 42161 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Arbitrum-selector"
  >
    <img
      alt="Arbitrum"
      class="c1"
      src="arbitrum_logo.svg"
    />
    <div
      class="c2"
    >
      Arbitrum
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 42220 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Celo-selector"
  >
    <img
      alt="Celo"
      class="c1"
      src="celo_logo.svg"
    />
    <div
      class="c2"
    >
      Celo
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 43114 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Avalanche-selector"
  >
    <img
      alt="Avalanche"
      class="c1"
      src="avax_logo.svg"
    />
    <div
      class="c2"
    >
      Avalanche
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 44787 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Celo Alfajores-selector"
  >
    <img
      alt="Celo Alfajores"
      class="c1"
      src="celo_logo.svg"
    />
    <div
      class="c2"
    >
      Celo Alfajores
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 80001 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Polygon Mumbai-selector"
  >
    <img
      alt="Polygon Mumbai"
      class="c1"
      src="polygon-matic-logo.svg"
    />
    <div
      class="c2"
    >
      Polygon Mumbai
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 84531 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c1 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="undefined-selector"
  >
    <div
      class="c1"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 421613 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Arbitrum Goerli-selector"
  >
    <img
      alt="Arbitrum Goerli"
      class="c1"
      src="arbitrum_logo.svg"
    />
    <div
      class="c2"
    >
      Arbitrum Goerli
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;

exports[`ChainSelectorRow should match snapshot for chainId 11155111 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: none;
  border: none;
  border-radius: 12px;
  color: #222222;
  cursor: pointer;
  display: grid;
  grid-template-columns: min-content 1fr min-content;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 20px;
  opacity: 1;
  padding: 10px 8px;
  text-align: left;
  -webkit-transition: 250ms ease background-color;
  transition: 250ms ease background-color;
  width: 240px;
}

.c0:hover {
  background-color: #22222212;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  font-size: 16px;
  font-weight: 485;
}

.c3 {
  grid-column: 3;
  grid-row: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 20px;
}

.c1 {
  height: 20px;
  width: 20px;
  margin-right: 12px;
}

@media only screen and (max-width:640px) {
  .c0 {
    width: 100%;
  }
}

<div>
  <button
    class="c0"
    data-testid="Sepolia-selector"
  >
    <img
      alt="Sepolia"
      class="c1"
      src="ethereum-logo.png"
    />
    <div
      class="c2"
    >
      Sepolia
    </div>
    <div
      class="c3"
    />
  </button>
</div>
`;
